linux/Platform.c

```diff
@@ -662,6 +673,91 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
    return true;
 }
 
+void Plateform_getBacktraces(Vector* frames, const Process* process, char **error) {
+   Vector_prune(frames);
+#ifdef HAVE_LIBUNWIND_PTRACE
+   unw_addr_space_t addrSpace = unw_create_addr_space(&_UPT_accessors, 0);
+   if (!addrSpace) {
+      xAsprintf(error, "Unable to init libunwind.");
+      return;
+   }
+
+   const pid_t pid = Process_getPid(process);
+
+   if (pid <= 0) {
+      xAsprintf(error, "Unable to get the pid");
+      goto addr_space_error;
+   }
+
+   if (ptrace(PTRACE_ATTACH, pid, 0, 0) == -1) {
+      xAsprintf(error, "ptrace: %s", strerror(errno));
+      goto addr_space_error;
+   }
+   wait(NULL);
+
+   struct UPT_info* context = _UPT_create(pid);
+   if (!context) {
+      xAsprintf(error, "Unable to init backtrace panel.");
+      goto ptrace_error;
+   }
+
+   unw_cursor_t cursor;
+   int ret = unw_init_remote(&cursor, addrSpace, context);
+   if (ret < 0) {
+      xAsprintf(error, "libunwind cursor: ret=%d", ret);
+      goto context_error;
+   }
+
+   int index = 0;
+   do {
+      char procName[256] = "?";
+      unw_word_t offset;
+      unw_word_t pc;
+
+      if (unw_get_proc_name(&cursor, procName, sizeof(procName), &offset) == 0) {
+         ret = unw_get_reg(&cursor, UNW_REG_IP, &pc);
+         if (ret < 0) {
+            xAsprintf(error, "unable to get register rip : %d", ret);
+            break;
+         }
+
+         Frame* frame = Frame_new();
+         frame->index = index;
+         frame->address = pc;
+         frame->offset = offset;
+         frame->isSignalFrame = unw_is_signal_frame(&cursor);
+#ifdef HAVE_LIBIBERTY
+         char* demangledName = cplus_demangle(procName,
```

Comment:
I don't think it's necessary to implement a switch between mangled and unmangled view

Reply:
I don't think it would hurt by keeping both the mangled and demangled names internally (in memory), even when you are not implementing the switch right now.

Reply:
I just discovered `eventHandler` from `Panel` 😄 . I'll implement it.